tests/typerel/t15563.nim

```diff
@@ -0,0 +1,9 @@
+discard """
+  matrix: "--gc:refc; --gc: arc"
+"""
+
+proc hello(x: varargs[string]) =
+  var s: seq[string]
+  s.add x
+
+hello("123")
```

Comment:
```nim
template main =
  proc hello(x: varargs[string]): seq[string] =
    var s: seq[string]
    s.add x
    s

  doAssert hello() == @[]
  doAssert hello("a1") == @["a1"]
  doAssert hello("a1", "a2") == @["a1", "a2"]

static: main()
main()
```
and move to something like `tests/system/tvarargs.nim` (so more tests can be added there) 

+ the usual js, c targets

Reply:
js + arc?